pkg/sources/source_manager.go

```diff
@@ -232,7 +249,10 @@ func (s *SourceManager) runWithoutUnits(ctx context.Context, handle handle, sour
 		defer wg.Done()
 		for chunk := range ch {
 			report.ReportChunk(nil, chunk)
-			_ = common.CancellableWrite(ctx, s.outputChunks, chunk)
+			if err := common.CancellableWrite(ctx, s.outputChunks, chunk); err != nil {
+				report.ReportError(Fatal{err})
```

Comment:
What makes this fatal?

Reply:
I have somewhat arbitrarily chosen to make context cancellations fatal errors so that they will surface as an error from a run, unlike errors we normally encounter but continue scanning.

Reply:
question: What happens if common.CancellableWrite encounters an error?

Looks like the error is reported by calling report.ReportError(Fatal{err}), and then the goroutine is immediately exited with return.

A potential problem here is that runWithoutUnits is not directly aware of this change. The error is reported via report.ReportError, but unless ReportError has side effects that the main function checks (which i hope isn't the case), I think this fxn will continue running.

In which case the source.Chunks function might continue to send chunks to the channel, but there is no longer a consumer for the channel since the goroutine has exited. I think?

Reply:
Maybe use an error channel to communicate errors from the goroutine back to the main function?